include/git2/repository.h

```diff
@@ -851,6 +852,17 @@ GIT_EXTERN(const char *) git_repository_get_namespace(git_repository *repo);
  */
 GIT_EXTERN(int) git_repository_is_shallow(git_repository *repo);
 
+/**
+ * Determine the shallow roots of the repository
+ *
+ * This oidarray is owned by the library. Do not free it.
+ *
+ * @param out An array of shallow oids.
+ * @param repo The repository
+ * @return 0 on success, an error otherwise.
+ */
+GIT_EXTERN(int) git_repository_shallow_roots(git_oidarray *out, git_repository *repo);
```

Comment:
We're probably unable to mark `git_oidarray` as const, right? Maybe we should instead just return a `git_grafts` structure? After all, that's how shallow roots are implemented.